### PR TITLE
[UI] App Frame with Sidebar

### PR DESCRIPTION
**Pull request by bhongy**
### What
- implements the new layout (sidebar) design for the `web` app

https://user-images.githubusercontent.com/3942264/215314579-90e3f623-3109-4626-96e2-a8fbd1f2a3d5.mov

How the menu behaves in a short screen.

https://user-images.githubusercontent.com/3942264/215315768-a273406b-3f7c-4068-8552-b2c807cf974d.mov

### CODE REVIEW

File: apps/web/components/EnhancedActiveLink.tsx
Comment (bhongy) [inline apps/web/components/EnhancedActiveLink.tsx:1]: no longer needed, see `NavLink`

File: apps/web/components/Layout.tsx
Comment (bhongy) [inline apps/web/components/Layout.tsx:1]: Not used; forget to removed in a previous PR.

File: apps/web/components/PageHeader.tsx
Comment (bhongy) [inline apps/web/components/PageHeader.tsx:11]: @iwasrobbed I made this a bit roomier than the design. Please let me know what you think.
image

File: apps/web/components/icons/DatabaseIcon.tsx
Comment (bhongy) [inline apps/web/components/icons/DatabaseIcon.tsx:8]: need to keep this for this glyph

File: apps/web/components/icons/SvgIcon.tsx
Comment (bhongy) [inline apps/web/components/icons/SvgIcon.tsx:9]: We can pass `props['aria-hidden'] = 'false'` to overwrite it.

File: apps/web/layouts/PageLayout/AuthLayout.tsx
Comment (bhongy) [inline apps/web/layouts/PageLayout/AuthLayout.tsx:1]: Moved to `apps/web/layouts/PageLayout/AuthLayout/AuthLayout.tsx`

File: apps/web/tailwind.config.js
Comment (bhongy) [inline apps/web/tailwind.config.js:119]: no longer needed with the removal of `EnhancedActiveLink`

File: apps/web/tailwind.config.js
Comment (bhongy) [inline apps/web/tailwind.config.js:27]: Need to provide the mapping to the theme so we can do `bg-venice-black-300` or `bg-venice-black-500`. `bg-venice-black` will use the `DEFAULT` i.e. maintains the same behavior.

We'll add more overtime as we need it.

File: apps/web/tailwind.config.js
Comment (bhongy) [inline apps/web/tailwind.config.js:29]: Needs to enable this to support `fill-current`, so we can style SVG icon colors using the container/parent style.

File: apps/web/tailwind.config.js
Comment (bhongy) [inline apps/web/tailwind.config.js:22]: Don't need this because anything passes to `extend.colors` is available for this already.

image

https://tailwindcss.com/docs/customizing-colors#using-custom-colors

File: apps/web/tailwind.config.js
Comment (bhongy) [inline apps/web/tailwind.config.js:30]: Don't need this because anything passes to `extend.colors` is available for this already.

image

https://tailwindcss.com/docs/customizing-colors#using-custom-colors

File: apps/web/styles/themes.js
Comment (bhongy) [inline apps/web/styles/themes.js:16]: match design

File: apps/web/public/venice-logo.svg
Comment (bhongy) [inline apps/web/public/venice-logo.svg:1]: rename and optimize svg

File: apps/web/pages/export.tsx
Comment (bhongy) [inline apps/web/pages/export.tsx:4]: To implement in a follow-up PR.

File: apps/web/pages/api-access.tsx
Comment (bhongy) [inline apps/web/pages/api-access.tsx:4]: To implement in a follow-up PR.

File: apps/web/layouts/PageLayout/PageLayout.tsx
Comment (bhongy) [inline apps/web/layouts/PageLayout/PageLayout.tsx:15]: Solve noisy error: “Title element received an array with more than 1 element as children.” https://modthefuture.com/fix-next-js-title-element-received-warning-react/

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (bhongy) [inline apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx:19]: Shall we change this to `/connections` (to do in a follow-up PR)?

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (bhongy) [inline apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx:33]: This will 404 for now.

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (bhongy) [inline apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx:36]: This will 404 for now.

File: apps/web/layouts/PageLayout/AuthLayout/NavLink.tsx
Comment (bhongy) [inline apps/web/layouts/PageLayout/AuthLayout/NavLink.tsx:14]: @iwasrobbed active link looks like this, wdyt?
image

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (tonyxiao) [reply apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx:18]: Yea I think we should def. keep the name & href the same either way :)

File: apps/web/pages/export.tsx
Comment (tonyxiao) [reply apps/web/pages/export.tsx:4]: We should keep the navigation for the current data-explorer page then its replacement is ready?

**Review by tonyxiao — APPROVED** [review]
Looks good, merge away after fixing the vercel deploy.

File: apps/web/components/HelpButton.tsx
Comment (iwasrobbed) [inline apps/web/components/HelpButton.tsx:7]: Nit: should be capital case `HELP` for this specific button

```suggestion
      <span className="text-xs uppercase">Help</span>
```

File: apps/web/components/PageHeader.tsx
Comment (iwasrobbed) [reply apps/web/components/PageHeader.tsx:11]: Looks good to me, just want to see it deployed since the fonts look a little large on your screen resolution (I have a zoomed out resolution to fit more on my screen, so might need to tweak font sizes)

File: apps/web/components/icons/DatabaseIcon.tsx
Comment (iwasrobbed) [reply apps/web/components/icons/DatabaseIcon.tsx:8]: Nit: would put this as a comment in code

File: apps/web/layouts/PageLayout/AuthLayout/NavLink.tsx
Comment (iwasrobbed) [reply apps/web/layouts/PageLayout/AuthLayout/NavLink.tsx:14]: Looks great!  I'll have a closer look once the Vercel deploy is fixed and I can see it in use

The only 🤔 is the left nav looks wider than in the designs so we may want to tweak that to be more narrow (again something we can look at once deployed)

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (iwasrobbed) [reply apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx:18]: Yes, please change to `/connections` and let's keep naming consistent

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (iwasrobbed) [reply apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx:32]: This should point to https://docs.venice.is/support/contact-us (new window)

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (iwasrobbed) [reply apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx:35]: This should point to https://docs.venice.is (new window)

File: apps/web/pages/export.tsx
Comment (iwasrobbed) [reply apps/web/pages/export.tsx:4]: Agreed, would keep the nav for the old page for now so we can keep using it until the new page is ready

**Review by iwasrobbed — APPROVED** [review]
Looks great, just want to feel it out once deployed 🙏  (we can tweak in follow-up merges as well)

**Comment by tonyxiao** [comment]
> Looks great, just want to feel it out once deployed 🙏 (we can tweak in follow-up merges as well)

@iwasrobbed every PR (in fact every commit) actually get deployed by Vercel to a unique URL. you'd be able to feel it out right now except for the fact that the deployment has failed and we need to update the PR :) Normally no need to wait for merge to check out.

**Comment by tonyxiao** [comment]
oh wait nvm scratch my previous comment. I think i misunderstood you.

File: apps/web/layouts/PageLayout/AuthLayout/NavLink.tsx
Comment (bhongy) [reply apps/web/layouts/PageLayout/AuthLayout/NavLink.tsx]: Good eye! Sorry I forgot to update that one. Corrected 538331e

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (bhongy) [reply apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx:18]: Will do in the next PR.

File: apps/web/components/icons/DatabaseIcon.tsx
Comment (bhongy) [reply apps/web/components/icons/DatabaseIcon.tsx:9]: Thanks, Rob! 4d8fedf

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (bhongy) [reply apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx]: 🙏 7e4788b

File: apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx
Comment (bhongy) [reply apps/web/layouts/PageLayout/AuthLayout/Sidebar.tsx]: 🙏 7e4788b

File: apps/web/pages/export.tsx
Comment (bhongy) [reply apps/web/pages/export.tsx]: 3ad9a79

**Comment by bhongy** [comment]
@iwasrobbed r? you can see the preview deployment now.

**Review by iwasrobbed — APPROVED** [review]
👍 👍  from my end, looks great and already feels much better than the top nav we had! I just noticed two minor things (see Vercel comments) 

We can add hover states later, would ship 🚀  and move onto next thing

**Comment by bhongy** [comment]
@iwasrobbed addressed your feedback in Vercel preview. Good catch.

**Pull request merged**
